robota_core/repository.py

```diff
@@ -483,7 +513,13 @@ def compare(self, point_1: str, point_2: str) -> List[Diff]:
         """
         if not point_1 + point_2 in self._diffs:
             gitlab_diffs = self.project.repository_compare(point_1, point_2)
-            robota_diffs = [Diff(diff, "gitlab") for diff in gitlab_diffs["diffs"]]
+
+            # base url for the diff
+            diff_url = f'{self.project_url}/-/compare/{point_1}...{point_2}'
+
+            # we calculate the page in the diff that this particular file will be on
+            robota_diffs = [Diff(diff, "gitlab", diff_url=diff_url, page=(i // GITLAB_DIFF_FILES_PER_PAGE + 1)) for i, diff in enumerate(gitlab_diffs["diffs"])]
```

Comment:
It will probably come back to bite us, but we have tended to just return everything for RoboTA Core without worrying about paging.  So one option would be for you to do the same here, and then we can look in a more systematic way across the code base to bring paging in properly?  Happy to discuss.

Reply:
Paging is only used on Gitlab diffs. Do you propose we simply link to the diff page and not worry about the anchor?

Reply:
Ideally, we would ask GitLab to return the full result without paging, and then we are guaranteed to find the anchor in what is returned.

Reply:
There isn't a way around the pagination on the web UI. I found an [issue](https://gitlab.com/gitlab-org/gitlab/-/issues/328727) tracking this.